### DISCUSSION TITLE
Buffer overruns & CPU overloads

### DISCUSSION THREAD

**Discussion by SpeedyCharly** [savonet/liquidsoap#2148] [Q&A]
Observed a re-occurrence of buffer overruns and excessive CPU usage for a few days while I was on rolling release #683ddf ;

2022/01/06 10:18:05 [clock.main:2] We must catchup 0.13 seconds!
2022/01/06 10:20:06 [clock.main:2] We must catchup 1.01 seconds!
2022/01/06 10:20:07 [clock.main:2] We must catchup 1.11 seconds!
2022/01/06 10:23:05 [clock.main:2] We must catchup 0.27 seconds!
2022/01/06 10:25:06 [clock.main:2] We must catchup 0.72 seconds!
2022/01/06 10:29:05 [clock.main:2] We must catchup 0.03 seconds!
2022/01/06 10:30:04 [clock.main:2] We must catchup 1.01 seconds!
2022/01/06 10:30:05 [clock.main:2] We must catchup 1.19 seconds!
2022/01/06 10:30:06 [clock.main:2] We must catchup 1.42 seconds!
2022/01/06 10:30:07 [clock.main:2] We must catchup 1.84 seconds!
2022/01/06 10:30:08 [clock.main:2] We must catchup 1.81 seconds!
2022/01/06 10:30:09 [clock.main:2] We must catchup 2.06 seconds!
2022/01/06 10:30:10 [clock.main:2] We must catchup 1.71 seconds!
2022/01/06 10:30:12 [clock.main:2] We must catchup 0.21 seconds!
2022/01/06 10:31:05 [clock.main:2] We must catchup 0.30 seconds!
2022/01/06 10:32:04 [clock.main:2] We must catchup 1.02 seconds!
2022/01/06 10:33:41 [clock.main:2] We must catchup 0.65 seconds!
2022/01/06 10:33:43 [clock.main:2] We must catchup 1.01 seconds!
2022/01/06 10:33:46 [clock.main:2] We must catchup 0.53 seconds!
2022/01/06 10:33:50 [clock.main:2] We must catchup 0.40 seconds!
2022/01/06 10:33:58 [harbor:4] New client on port 8005: 24.50.66.199

This problem resolved itself as soon as I updated to #abaa4a0, but I see a lot of occurrences as below ;

2022/01/07 12:00:19 [amplify_0:4] Overriding amplification: 1.000000.
2022/01/07 12:00:19 [amplify_2:4] Overriding amplification: 1.000000.
2022/01/07 12:00:19 [process:4] Starting process
2022/01/07 12:00:19 [process:4] Closing process's stdin
2022/01/07 12:00:19 [process:4] Process exited with code 0
2022/01/07 12:00:19 [process:4] Cleaning up process
2022/01/07 12:00:19 [lang:3] AzuraCast Feedback Response: OK
2022/01/07 12:00:20 [buffer_417:4] Buffer emptied, buffering needed.
2022/01/07 12:00:20 [buffer_416:4] Buffer emptied, buffering needed.
2022/01/07 12:00:20 [crossfade_0:4] Buffering end of track...
2022/01/07 12:00:21 [crossfade_0:4] More buffering will be needed.
2022/01/07 12:00:36 [input_streamer:4] End of track.
2022/01/07 12:00:36 [live_fallback:3] Switch to map_metadata_5 with transition.
2022/01/07 12:00:36 [amplify_0:4] End of the current overriding.
2022/01/07 12:00:36 [amplify_2:4] End of the current overriding.

Does that mean that the buffering issue isn't quite resolved yet?

**Comment by toots** [comment]
It looks like something is blocking the main streaming thread when a user connects. Is there an API call happening on login?

**Reply by SpeedyCharly** [reply]
Not sure what u mean with API call ... will send a typical login log...

**Reply by toots** [reply]
I was thinking a AzuraCast API call. I think this issue should be better debugged there first.

**Reply by SpeedyCharly** [reply]
Here are my logs 1st = buffer issue during AutoDJ operation ;
[Buffer issue in AutoDJ.txt](https://github.com/savonet/liquidsoap/files/7836429/Buffer.issue.in.AutoDJ.txt)
This before 1st morning DJ 

First DJ loging in this AM ;
[First Login.txt](https://github.com/savonet/liquidsoap/files/7836434/First.Login.txt)

DJ switchover ;
[Dj Switch.txt](https://github.com/savonet/liquidsoap/files/7836435/Dj.Switch.txt)

Sorry I still have no idea what is meant by API calls; not dense, just not a Linuxian here...
